hal/include/HAL/SerialPort.h

```diff
@@ -9,31 +9,42 @@
 
 #include <stdint.h>
 
+enum HAL_SerialPort : int32_t {
```

Comment:
We changed all of these in the HAL a few months ago to do this. At the time we determined that needing actual C compiler compatibility was not necessary, just a C compatible API for interop.